apollo-router/src/plugins/telemetry/dynamic_attribute.rs

```diff
@@ -0,0 +1,159 @@
+use std::collections::HashMap;
+
+use tracing_subscriber::layer::Context;
+use tracing_subscriber::registry::LookupSpan;
+use tracing_subscriber::Layer;
+use tracing_subscriber::Registry;
+
+use super::ROUTER_SPAN_NAME;
+use super::SUBGRAPH_SPAN_NAME;
+use super::SUPERGRAPH_SPAN_NAME;
+
+#[derive(Debug)]
+pub(crate) enum LogAttributes {
+    Router(HashMap<String, String>),
+    Supergraph(HashMap<String, String>),
+    Subgraph(HashMap<String, String>),
+}
```

Comment:
if all variants have the same content, it may be nicer with a flag:

```suggestion
pub(crate) enum SpanKinf {
  Router,
  Supergraph,
  Subgraph
}
pub(crate) struct LogAttributes {
  kind: SpanKind,
  attributes: HashMap<String, String>,
}
```